v5: Modular Installation Architecture

parthsidpara opened the pull request:
This PR moves PiOSK from a `git clone` based installation to a robust, and self-contained package model.

### Key Changes:

**1. `bootstrap.sh`: The New Entry Point**
The primary way to interact with PiOSK is now through a single bootstrap script. Instead of piping `setup.sh` from the repo, users will now run:
`curl -sSL https://code.debs.io/piosk/bootstrap.sh | sudo bash -s -- [command]`

**2. Self Contained Release Packages**
- Users download a `.tar.gz` containing only the essential files.
- The target system is no longer cluttered with unnecessary files like the `www/` website code, `README.md`, or .git history.

**3. Script Refactoring**
- `scripts/` directory has been refactored to separate directories
- `scripts/setup/`: Contains all scripts related to managing the installation lifecycle (`install.sh`, `update.sh`, `cleanup.sh`, `backup.sh`, `common.sh`).
- `scripts/runtime/`: Contains the scripts executed by the systemd services during normal operation (`runner.sh`, `switcher.sh`).

**4. Update Build and Release Workflow**
The `.github/workflows/build-release.yml` workflow has been updated to support this new model:
- It now assembles a complete release package (`.tar.gz`) containing the `dashboard/`, `scripts/`, `services/`, `config.json.sample` and the compiled binary.
- Release assets are automatically versioned based on the git tag.
- New releases are created as "pre-releases" by default, allowing for a final verification step before being manually promoted to "Latest".

parthsidpara committed:
feat: add bootstrap script

parthsidpara committed:
refactor: move runner and switcher scripts to scripts/runtime

parthsidpara committed:
refactor: remove old setup script

parthsidpara committed:
refactor: move cleanup script to scripts/setup

parthsidpara committed:
feat: add setup scripts for installation, backup and update

parthsidpara committed:
feat: update old cleanup script

parthsidpara committed:
refactor: update script paths in runner and switcher templates

parthsidpara committed:
feat: update build-release workflow for new modular architecture

debloper commented on scripts/setup/install.sh:
```diff
@@ -0,0 +1,71 @@
+# 2. INSTALL RUNTIME DEPENDENCIES
+msg "$INFO" "Installing runtime dependencies (wtype, chromium-browser)..."
+apt-get update
+apt-get install -y wtype chromium-browser
```
Ignore.

parthsidpara committed:
refactor: remove stdout supression from apt-get command

debloper commented on .github/workflows/build-release.yml:
```diff
@@ -66,9 +79,6 @@ jobs:
     if: startsWith(github.ref, 'refs/tags/v')
```
Ensure workflow-dispatch doesn't break this step.

debloper reviewed (changes requested):
Needs some fixes; mainly around workflow and bootstrap.sh

parthsidpara committed:
refactor(workflow): simplify build matrix

parthsidpara committed:
refactor(workflow): remove version from release filename

parthsidpara committed:
feat(workflow): add --version flag

Adds a  flag to the  executable to allow users and scripts to identify the build version.

parthsidpara committed:
refactor: remove version and jq dependency from download logic in bootstrap script

parthsidpara committed:
refactor(workflow): update and simplify package naming

parthsidpara committed:
remove version.ts file

parthsidpara committed:
remove --version feature entirely

debloper reviewed (approved):
Looks good enough to me; we can trim off the issues iteratively.
